src/server/scripts/Outland/BlackTemple/boss_mother_shahraz.cpp

```diff
@@ -212,7 +212,26 @@ class spell_mother_shahraz_fatal_attraction : public SpellScript
 
     void SetDest(SpellDestination& dest)
     {
-        dest.Relocate(GetCaster()->GetRandomNearPosition(50.0f));
+        float constexpr minDist = 30.0f;
+        float constexpr maxDist = 48.0f;
+
+        // Randomize the teleportation distance from the caster
+        float const teleportDist = frand(minDist, maxDist);
+
+        // Initialize a first destination
+        Position teleportDest = GetCaster()->GetRandomNearPosition(teleportDist);
+
+        // Ensure that the destination is not too close to the caster.
+        // Add a check for LOS, to ensure to not be teleported under the map
+        while (teleportDest.GetExactDist(GetCaster()) < minDist ||
```

Comment:
Is there a way for us to avoid the while here?

Reply:
No, but Can you put this PR in WIP ? I'm not pleased to this PR because it's avoid to be near a wall from the boss but not from the other wall. I think that we a fonction to check if there are a wall in a radius. I will check this later.